crates/next-error-code-swc-plugin/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+                    span: new_error_expr.span,
+                    callee: Callee::Expr(Box::new(Expr::Member(MemberExpr {
+                        span: new_error_expr.span,
+                        obj: Box::new(Expr::Ident(Ident::new(
+                            "Object".into(),
+                            new_error_expr.span,
+                            Default::default(),
+                        ))),
+                        prop: MemberProp::Ident("assign".into()),
+                    }))),
+                    args: vec![
+                        ExprOrSpread {
+                            spread: None,
+                            expr: Box::new(Expr::New(new_error_expr.clone())),
+                        },
+                        ExprOrSpread {
+                            spread: None,
+                            expr: Box::new(Expr::Object(ObjectLit {
+                                span: new_error_expr.span,
+                                props: vec![PropOrSpread::Prop(Box::new(Prop::KeyValue(
+                                    KeyValueProp {
+                                        key: PropName::Ident("__NEXT_ERROR_CODE".into()),
+                                        value: Box::new(Expr::Lit(Lit::Str(Str {
+                                            span: new_error_expr.span,
+                                            value: code.into(),
+                                            raw: None,
+                                        }))),
+                                    },
+                                )))],
+                            })),
+                        },
+                    ],
+                    type_args: None,
+                    ctxt: new_error_expr.ctxt,
+                });
+            }
+        }
+    }
+}
+
+#[plugin_transform]
+pub fn process_transform(
+    mut program: Program,
+    metadata: TransformPluginProgramMetadata,
+) -> Program {
+    fn parse_config(metadata: TransformPluginProgramMetadata) -> (String, String, String) {
+        let config_str = metadata
+            .get_transform_plugin_config()
+            .unwrap_or_else(|| panic!("Failed to get transform plugin config"))
+            .to_string();
+
+        let config: serde_json::Value = serde_json::from_str(&config_str)
+            .unwrap_or_else(|e| panic!("failed to parse config: {}, config: {}", e, config_str));
+
+        (
+            config["commitHash"]
+                .as_str()
+                .unwrap_or_else(|| panic!("commitHash not found in config: {}", config_str))
+                .to_string(),
+            config["filePath"]
+                .as_str()
+                .unwrap_or_else(|| panic!("filePath not found in config: {}", config_str))
+                .to_string(),
+            config["mode"]
+                .as_str()
+                .unwrap_or_else(|| panic!("mode not found in config: {}", config_str))
+                .to_string(),
+        )
+    }
```

Comment:
hmm i think the standard approach here is define a struct and let serde derive this code? something like
```rs
#[derive(Serialize, Deserialize, Debug)]
struct Config {
    commitHash: String,
    filePath: String,
    mode: Mode // see my other comment about using an enum for this
}
```
https://serde.rs/#data-structures